docs_src/design/legacy-requirements/device-service.md

```diff
@@ -0,0 +1,157 @@
+Port | Int | Port on which to accept the device service's REST API. The assigned port for experimental / in-development device services is 49999.
+Timeout | Int | Time (in milliseconds) to wait between attempts to contact core-data and core-metadata when starting up.
+ConnectRetries | Int | Number of times to attempt to contact core-data and core-metadata when starting up.
+StartupMsg | String | Message to log on successful startup.
+CheckInterval | String | The checking interval to request if registering with Consul
+ServerBindAddr | String | The interface on which the service's REST server should listen. By default the server listens on all available interfaces.
+
+#### Clients section
+
+Defines the endpoints for other microservices in an EdgeX system.
+Not required when using Registry.
+
+##### Data
+Option | Type | Notes
+:--- | :--- | :---
+Host | String | Hostname on which to contact the core-data service.
+Port | Int | Port on which to contact the core-data service.
+
+##### Metadata
+
+Option | Type | Notes
+:--- | :--- | :---
+Host | String | Hostname on which to contact the core-metadata service.
+Port | Int | Port on which to contact the core-metadata service.
+
+#### Device section
+
+Option | Type | Notes
+:--- | :--- | :---
+DataTransform | Bool | For enabling/disabling transformations on data between the device and EdgeX. Defaults to true (enabled).
+Discovery/Enabled | Bool | For enabling/disabling device discovery. Defaults to true (enabled).
+Discovery/Interval | Int | Time between automatic discovery runs, in seconds. Defaults to zero (do not run discovery automatically).
+MaxCmdOps | Int | Defines the maximum number of resource operations that can be sent to the driver in a single command.
+MaxCmdResultLen | Int | Maximum string length for command results returned from the driver.
+UpdateLastConnected | Bool | If true, update the LastConnected attribute of a device whenever it is successfully accessed. Defaults to false.
+
+#### Logging section
+
+Option | Type | Notes
+:--- | :--- | :---
+LogLevel | String | Sets the logging level. Available settings in order of increasing severity are: `TRACE`, `DEBUG`, `INFO`, `WARNING`, `ERROR`.
+
+#### Driver section
+
+This section is for driver-specific options. Any configuration specified here will be passed to the driver implementation during initialization.
+
+## Push Events
+
+The SDK should implement methods for generating Events other than on
+receipt of device GET requests. The AutoEvent mechanism provides for
+generating Events at fixed intervals. The asynchronous event queue
+enables the device service to generate events at arbitrary times,
+according to implementation-specific logic.
+
+### AutoEvents
+
+Each device may have as part of its definition in Metadata a number of `AutoEvents` associated with it. An `AutoEvent` has the following fields:
+
+* **resource**: the name of a deviceresource or resourcecommand indicating what to read.
```

Comment:
we renamed resourcecommand to device command

Reply:
Yes. Updated, thanks